homework-submission/React_HW2/todoApp2.jsx

```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+// ContentDynamicTodo component is for map function to use
+const ContentDynamicTodo = ({ description, deadline, className, deleteItem }) => {
+  return (
+    <tr className={className} onClick={deleteItem}>
+      <td className="content"> {description} </td>
+      <td className="content"> {deadline} </td>
+      <td className="content"> {className} </td>
+    </tr>
+  );
+};
+
+// DynamicList where map function is
+const DynamicList = ({ list, deleteItem }) => {
+  const runTodoList = list.map(todo => (
+    <ContentDynamicTodo
+      key={todo.id}
+      description={todo.description}
+      deadline={todo.deadline}
+      className={todo.done ? '√' : 'x'}
+      deleteItem={() => {
+        deleteItem(todo.id);
+      }}
+    />
+  ));
+  return runTodoList;
+};
+
+// Here is our main Application component as a class
+class AppTodo2 extends React.Component {
+  //TodoList which contains new item as state
+  constructor(props) {
+    super(props);
+    this.state = {
+      todoList: [
+        {
+          id: 1,
+          description: 'Get out of bed',
+          deadline: '2017-09-11',
+          done: true,
+        },
+        {
+          id: 2,
+          description: 'Brush teeth',
+          deadline: '2017-09-10',
+          done: false,
+        },
+        {
+          id: 3,
+          description: 'Eat breakfast',
+          deadline: '2017-09-09',
+          done: false,
+        },
+      ],
+    };
+  }
+
+  newTodo = {
+    id: '',
+    description: 'Description',
+    deadline: 'No Deadline Entry',
+    done: false,
+  };
+
+  NewTodoDescriptionAndDeadline = () => {
+    return (
+      <div>
+        <div>
+          <label> Description </label>
+          <div id="containerDescriptionInput">
+            <input
+              type="text"
+              onChange={event => {
+                const inputValue = event.target.value;
+                this.setState({
+                  newTodo: {
```

Comment:
You don't have a `newTodo` item in your state, this will make it harder to debug and harder to follow in some cases.